pkg/sidecar/appconf.go

```diff
@@ -114,25 +124,78 @@ func getIdentityConfigs(id int, reportHost string) (*ini.File, error) {
 	return cfg, nil
 }
 
-func getUtilityUserConfigs(user, pass string) (*ini.File, error) {
+func getInitFileConfigs(filePath string) (*ini.File, error) {
 	cfg := ini.Empty()
 	mysqld := cfg.Section("mysqld")
 
-	if _, err := mysqld.NewKey("utility-user", fmt.Sprintf("%s@%%", user)); err != nil {
-		return nil, err
-	}
-	if _, err := mysqld.NewKey("utility-user-password", pass); err != nil {
+	if _, err := mysqld.NewKey("init-file", filePath); err != nil {
 		return nil, err
 	}
-	if _, err := mysqld.NewKey("utility-user-schema-access", "mysql"); err != nil {
-		return nil, err
+
+	return cfg, nil
+}
+
+func initFileQuery(cfg *Config) []byte {
+	queries := []string{
+		"SET @@SESSION.SQL_LOG_BIN = 0;",
 	}
-	if _, err := mysqld.NewKey("utility-user-privileges",
-		"SELECT,INSERT,UPDATE,DELETE,CREATE,DROP,GRANT,ALTER,SHOW DATABASES,SUPER,CREATE USER,"+
-			"PROCESS,RELOAD,LOCK TABLES,REPLICATION CLIENT,REPLICATION SLAVE",
-	); err != nil {
-		return nil, err
+
+	// configure operator utility user
+	queries = append(queries, createUserQuery(cfg.OperatorUser, cfg.OperatorPassword, "%",
+		[]string{"SUPER", "SHOW DATABASES", "PROCESS", "RELOAD", "CREATE", "SELECT"}, "*.*",
+		[]string{"ALL"}, fmt.Sprintf("%s.*", toolsDbName)))
+
+	// configure orchestrator user
+	queries = append(queries, createUserQuery(cfg.OrchestratorUser, cfg.OrchestratorPassword, "%",
+		[]string{"SUPER", "PROCESS", "REPLICATION SLAVE", "REPLICATION CLIENT", "RELOAD"}, "*.*",
+		[]string{"SELECT"}, "mysql.slave_master_info"))
+
+	// configure replication user
+	queries = append(queries, createUserQuery(cfg.ReplicationUser, cfg.ReplicationPassword, "%",
+		[]string{"SELECT", "PROCESS", "RELOAD", "LOCK TABLES", "REPLICATION CLIENT", "REPLICATION SLAVE"}, "*.*"))
+
+	// configure metrics exporter user
+	queries = append(queries, createUserQuery(cfg.MetricsUser, cfg.MetricsPassword, "127.0.0.1",
+		[]string{"SELECT", "PROCESS", "REPLICATION CLIENT"}, "*.*"))
+	queries = append(queries, fmt.Sprintf("ALTER USER %s@'127.0.0.1' WITH MAX_USER_CONNECTIONS 3;", cfg.MetricsUser))
+
+	// configure heartbeat user
+	// because of pt-heartbeat make sure not to have ALL or SUPER privileges:
+	// https://github.com/percona/percona-toolkit/blob/e85ce15ef24bc4614b4d2f13792fa73583d68f8e/bin/pt-heartbeat#L6433
+	queries = append(queries, createUserQuery(cfg.HeartBeatUser, cfg.HeartBeatPassword, "127.0.0.1",
+		[]string{"CREATE", "SELECT", "DELETE", "UPDATE", "INSERT"}, fmt.Sprintf("%s.%s", toolsDbName, toolsHeartbeatTableName),
+		[]string{"REPLICATION CLIENT"}, "*.*"))
+
+	return []byte(strings.Join(queries, "\n"))
```

Comment:
Consider `queries` as a slice of queries which need to be appended with `;` and `\n`